### PR TITLE
Resolve warning during compilation for all rtl8721csm configs 

### PR DESCRIPTION
**Pull request by Kr159**
made necessary changes in os folder to resove the warnings

### CODE REVIEW

**Comment by edwakuwaku** [comment]
This changes is not specific to rtl8721csm. Please revise your PR title.

File: os/mm/umm_heap/umm_realloc.c [hunk 1]
```diff
@@ -137,7 +137,7 @@ void *realloc_at(int heap_index, void *oldmem, size_t size)
 
 FAR void *realloc(FAR void *oldmem, size_t size)
 {
-	int prev_heap_idx;
+       __attribute__ ((unused))	int prev_heap_idx;
```
Comment (edwakuwaku) [inline os/mm/umm_heap/umm_realloc.c:140]: Any reason you change like this?
It should be as below, please check again
```
#if !defined(CONFIG_APP_BINARY_SEPARATION)
int prev_heap_idx;
#endif
```

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c [hunk 2]
```diff
@@ -43,7 +43,7 @@ struct _tizenrt_timer_entry {
 };
 
 static _list _tizenrt_timer_table;
-static u8 _tizenrt_timer_table_init = 0;
```
Comment (edwakuwaku) [inline os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:46]: Please revert the changes in this file. We will make necessary changes

**Comment by edwakuwaku** [comment]
There is a typo in your commit description:
resove -> resolve

File: os/mm/umm_heap/umm_realloc.c [hunk 3]
```diff
@@ -137,7 +137,9 @@ void *realloc_at(int heap_index, void *oldmem, size_t size)
 
 FAR void *realloc(FAR void *oldmem, size_t size)
 {
-	int prev_heap_idx;
+	#ifndef CONFIG_APP_BINARY_SEPARATION
```
Comment (edwakuwaku) [inline os/mm/umm_heap/umm_realloc.c:140]: This alignment is wrong.. you should keep #if indicator to leftmost, please check other place usage of this...
Also, I'm not sure whether #ifndef is default usage. Normally, it should be either `#ifdef` or `#if !defined(CONFIG_XXX..)`
@sunghan-chang  could you please help to clarify?

**Comment by edwakuwaku** [comment]
Hi @Kr159 , for the changes regarding comments, you shouldn't push another commit, because those are within the same scope of your first commit...
Instead, you should amend your first commit and force push again... So you can keep relevant changes in just one commit..

File: os/mm/umm_heap/umm_realloc.c (hunk 3 above)
Comment (sunghan-chang) [reply os/mm/umm_heap/umm_realloc.c:140]: Agree two things @edwakuwaku said.

**Comment by r-prabu** [comment]
@Kr159 make sure for the review fixes amend the changes in the first commit.

**Review by r-prabu — CHANGES_REQUESTED** [review]
Need changes.

**Review by kishore-sn — APPROVED** [review]

**Comment by edwakuwaku** [comment]
Hi @Kr159 , the PR title is still not changed..

**Comment by Kr159** [comment]
> Hi @Kr159 , the PR title is still not changed..

> Hi @edwakuwaku , I changed PR Title from "**Resolve warnings for rtl8721csm**" to "**Resolve warning during compilation for all rtl8721csm configs**".
